lib/common/grpc-service.js

```diff
@@ -325,7 +325,8 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
 
     request: function() {
       return service[protoOpts.method](reqOpts, grpcOpts)
-        .on('status', function(status) {
+        .on('metadata', function() {
+          var status = this.received_status || { code: 0 };
```

Comment:
I think we should get the status from grpc if it's available, we just need to know the right way to do that.